### PR TITLE
MyBatis

### PR DESCRIPTION
**Pull request by Saranyar04**

### CODE REVIEW

File: src/main/java/org/example/mybatis/dao/AppointmentDAO.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import org.apache.ibatis.session.SqlSessionFactory;
+import org.example.interfaces.IAppointmentDAO;
+import org.example.models.hospital.Appointment;
+import org.example.util.MyBatisSqlFactory;
+
+import java.util.List;
+
+public class AppointmentDAO implements IAppointmentDAO {
+
+    private static SqlSessionFactory sqlSessionFactory;
```
Comment (ankochergin) [inline src/main/java/org/example/mybatis/dao/AppointmentDAO.java]: could you please explain why it's static?

File: src/main/java/org/example/Main.java [hunk 2]
```diff
@@ -3,16 +3,16 @@
...
+import org.example.mybatis.dao.*;
 
 import java.util.List;
 
 public class Main {
     private static final Logger LOGGER = (Logger) LogManager.getLogger(Main.class);
 
     public static void main(String[] args) {
-        AppointmentService appointmentService = new AppointmentService();
+        /*AppointmentService appointmentService = new AppointmentService();
```
Comment (ankochergin) [inline src/main/java/org/example/Main.java]: just replace JDBCDAO in your service layer with MyBatis DAOs and run commented code

File: src/main/java/org/example/mybatis/dao/AppointmentDAO.java (hunk 1 above)
Comment (Saranyar04) [reply src/main/java/org/example/mybatis/dao/AppointmentDAO.java]: As I explained in class I tried various things to fix the issue wit sql session factory in AppointmentDAO but did not change it back now deleted that.

File: .idea/.gitignore
Comment (ankochergin) [inline .idea/.gitignore:1]: remove .idea folder from git

File: logs/pharmacy.log
Comment (ankochergin) [inline logs/pharmacy.log:1]: please remove log file from git

File: src/main/java/org/example/Main.java [hunk 3]
```diff
@@ -2,52 +2,98 @@
...
-        int count = specializationService.getCountOfSpecialization();
-        LOGGER.info("No of Specializations : " + count);
+        ProcedureDAO procedureDAO = new ProcedureDAO();
+        procedureDAO.removeEntityByID(9);
+        /*Procedure procedure = new Procedure();
+        procedure.setProcedureID(10);
+        procedure.setName("Coronary Artery Bypass");
+        procedure.setCost(2330.00);
+        procedure.setDescription("Treat coronary artery disease");
+        procedureDAO.saveEntity(procedure);*/
```
Comment (ankochergin) [inline src/main/java/org/example/Main.java]: remove commented code if it's no longer used

File: src/main/java/org/example/mybatis/dao/AppointmentDAO.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import org.example.util.MyBatisSqlFactory;
+
+import java.util.List;
+
+public class AppointmentDAO implements IAppointmentDAO {
+
+    private SqlSessionFactory sqlSessionFactory = MyBatisSqlFactory.getSqlSessionFactory();
+    public AppointmentDAO() {
+
+    }
```
Comment (ankochergin) [inline src/main/java/org/example/mybatis/dao/AppointmentDAO.java]: empty constructor can be removed, it's default

File: src/main/java/org/example/parser/DOM.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.IOException;
+
+public class DOM {
```
Comment (ankochergin) [inline src/main/java/org/example/parser/DOM.java]:
```suggestion
public class DOMParser {
```

File: src/main/resources/db.properties [hunk 6]
```diff
@@ -1,4 +1,4 @@
 driver=com.mysql.cj.jdbc.Driver
 url=jdbc:mysql://localhost:3306/hospital
 user=root
-password=
\ No newline at end of file
+password=Thanusha-21
```
Comment (ankochergin) [inline src/main/resources/db.properties]: don't commit passwords

File: src/main/java/org/example/parser/DOM.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+    private static final Logger LOGGER = (Logger) LogManager.getLogger(DOM.class);
+
+    public static void main(String[] args) {
```
Comment (ankochergin) [inline src/main/java/org/example/parser/DOM.java]: please add validation of your XML file by XSD schema

**Review by ankochergin — COMMENTED** [review]
next time please don't combine multiple tasks in 1 PR

**Pull request merged**